blockchain/pending.go

```diff
@@ -122,6 +122,6 @@ func (p *PendingState) SetCompiledClassHash(classHash, compiledClassHash *felt.F
 
 // StateDiff returns the pending state's internal state diff. The returned object will continue to be
 // read and modified by the pending state.
-func (p *PendingState) StateDiff() *core.StateDiff {
-	return p.stateDiff
+func (p *PendingState) StateDiff() (*core.StateDiff, map[felt.Felt]core.Class) {
```

Comment:
Do you think it would be more clear if we had a separate `func (p *PendingState) NewClasses()` function?

Reply:
We can rename it to `StateDiffAndClasses`. Wdyt?

Reply:
Yeah that works too